src/NexusMods.App.UI/LeftMenu/Items/ApplyControlViewModel.cs

```diff
@@ -40,56 +49,72 @@ public ApplyControlViewModel(LoadoutId loadoutId, IServiceProvider serviceProvid
         _applyService = serviceProvider.GetRequiredService<IApplyService>();
         LaunchButtonViewModel = serviceProvider.GetRequiredService<ILaunchButtonViewModel>();
         LaunchButtonViewModel.LoadoutId = loadoutId;
+        
+        NewestLoadout = _loadoutRegistry.Get(_loadoutId) ??
+                       throw new ArgumentException("Loadout not found: " + _loadoutId);
+        
+        _gameInstallation = NewestLoadout.Installation;
 
-
-        _gameInstallation = _loadoutRegistry.Get(_loadoutId)?.Installation ??
-                            throw new ArgumentException("Loadout not found: " + _loadoutId);
-
-        (_lastAppliedLoadoutId, _lastAppliedRevisionId) = GetLastAppliedLoadout();
-
-        _newestLoadout = _loadoutRegistry.RevisionsAsLoadouts(loadoutId)
-            .ToProperty(this, vm => vm.NewestLoadout, scheduler: RxApp.MainThreadScheduler);
+        LastAppliedRevisionId = _applyService.GetLastAppliedLoadout(_gameInstallation) ??
+                                throw new ArgumentException("No last applied loadout found for: " +
+                                                            _gameInstallation);
         
         _applyReactiveCommand = ReactiveCommand.CreateFromTask(async () => await Apply());
 
         this.WhenActivated(disposables =>
         {
+            _loadoutRegistry.RevisionsAsLoadouts(loadoutId)
+                .OnUI()
+                .BindToVM(this, vm => vm.NewestLoadout)
+                .DisposeWith(disposables);
+
+            _applyService.LastAppliedRevisionFor(_gameInstallation)
+                .OnUI()
+                .BindToVM(this, vm => vm.LastAppliedRevisionId)
+                .DisposeWith(disposables);
```

Comment:
Use [OAPH](https://github.com/Nexus-Mods/NexusMods.App/blob/main/docs/development-guidelines/UICodingGuidelines.md#viewmodel-properties) instead of `BindToVM` to expose these two properties.

Reply:
I didn't use OAPH deliberately here to ensure that the update would happen immediately:

> It is important to note that OAPH will not set the property value immediately, but will rather schedule it on the provided or default scheduler. The [Reactive] property with .BindTo() should be used for business-critical code.

Second paragraph here:
https://www.reactiveui.net/docs/handbook/observable-as-property-helper

Reply:
And from our own coding guidelines:

> Always use ObservableAsPropertyHelper<T> (OAPH) to expose the latest values from an IObservable<T> that is async or runs on the task pool scheduler. The OAPH is scheduled by default meaning it doesn't set the value immediately.